scripts/test/system.sh

```diff
@@ -110,6 +110,25 @@ EOF
     echo "::endgroup::"
 }
 
+function test_subctl_recover_broker_info() {
+    # Rename old broker-info.subm to be able to compare it with the recovered broker-info.sum
+    echo "Renaming broker-info.subm to broker-info.subm.orig"
+    mv "${DAPPER_SOURCE}"/output/broker-info.subm "${DAPPER_SOURCE}"/output/broker-info.subm.orig
+    base64 -d "$DAPPER_SOURCE"/output/broker-info.subm.orig > "$DAPPER_SOURCE"/output/decoded_broker_info.subm.orig
+
+    _subctl recover-broker-info --context cluster1
+    cmp_rm_files
+
+    _subctl recover-broker-info --context cluster2
+    cmp_rm_files
+}
+
+function cmp_rm_files() {
+  base64 -d broker-info.subm > decoded_broker_info.subm
+  diff <(yq -P eval decoded_broker_info.subm) <(yq -P eval "$DAPPER_SOURCE"/output/decoded_broker_info.subm.orig)
+  rm -f broker-info.subm decoded_broker_info.subm
+}
```

Comment:
What I mean is something like this:

```suggestion
function test_subctl_recover_broker_info() {
    # Rename old broker-info.subm to be able to compare it with the recovered broker-info.sum
    echo "Renaming broker-info.subm to broker-info.subm.orig"
    mv "${DAPPER_SOURCE}"/output/broker-info.subm "${DAPPER_SOURCE}"/output/broker-info.subm.orig
    base64 -d "$DAPPER_SOURCE"/output/broker-info.subm.orig > "$DAPPER_SOURCE"/output/decoded_broker_info.subm.orig

  _subctl recover-broker-info "$@"
  base64 -d broker-info.subm > decoded_broker_info.subm
  diff <(yq -P eval decoded_broker_info.subm) <(yq -P eval "$DAPPER_SOURCE"/output/decoded_broker_info.subm.orig)
  rm -f broker-info.subm decoded_broker_info.subm
  
  mv "${DAPPER_SOURCE}"/output/broker-info.subm.orig "${DAPPER_SOURCE}"/output/broker-info.subm
}

... later in the file ...

test_subctl_recover_broker_info --context cluster1

test_subctl_recover_broker_info --context cluster2
```

Reply:
Either way is fine - as Mike said, it's just a system test - but Mike's approach seems a bit cleaner, ie it isolates the code in one function.

Reply:
This follows the model of `subctl_gather`.

Reply:
> This follows the model of `subctl_gather`.

and `test_subctl_diagnose_in_cluster`